ark/utils/plot_utils.py

```diff
@@ -11,15 +11,49 @@
 from skimage.exposure import rescale_intensity
 
 
-# plotting functions
+def preprocess_tif(predicted_contour, plotting_tif):
+    """Validates plotting_tif and preprocesses it accordingly
 
-def plot_overlay(predicted_contour, plotting_tif, alternate_contour=None, path=None, show=False):
+    Args:
+        predicted_contour (numpy.ndarray):
+            2D numpy array of labeled cell objects
+        plotting_tif (numpy.ndarray):
+            2D or 3D numpy array of imaging signal
+
+    Returns:
+        numpy.ndarray:
+            The preprocessed image
+    """
+
+    if len(plotting_tif.shape) == 2:
+        if plotting_tif.shape != predicted_contour.shape:
+            raise ValueError("plotting_tif and predicted_contour array dimensions not equal.")
+        else:
+            # convert RGB image with same data across all three channels
+            formatted_tif = np.stack((plotting_tif, plotting_tif, plotting_tif), axis=2)
+    elif len(plotting_tif.shape) == 3:
+        # can only support up to 3 channels
+        if plotting_tif.shape[2] > 3:
+            raise ValueError("max 3 channels of overlay supported, got {}".
+                             format(plotting_tif.shape))
+
+        # set first n channels of formatted_tif to plotting_tif (n = num channels in plotting_tif)
+        formatted_tif = np.zeros((plotting_tif.shape[0], plotting_tif.shape[1], 3),
+                                 dtype=plotting_tif.dtype)
+        formatted_tif[..., :plotting_tif.shape[2]] = plotting_tif
+    else:
+        raise ValueError("plotting tif must be 2D or 3D array, got {}".
+                         format(plotting_tif.shape))
+
+    return formatted_tif
+
+
+def create_overlay(predicted_contour, plotting_tif, alternate_contour=None):
     """Take in labeled contour data, along with optional mibi tif and second contour,
     and overlay them for comparison"
 
-    Plots the outline(s) of the mask(s) as well as intensity from plotting tif.  Predicted
-    contours are plotted in red, while alternate contours are plotted in white.  Plot is
-    saved as a TIF file in provided file path, if specified.
+    Generates the outline(s) of the mask(s) as well as intensity from plotting tif. Predicted
```

Comment:
typo here. Also, top-level description should be updated. I think segmentation_label is more clear than labeled contour data